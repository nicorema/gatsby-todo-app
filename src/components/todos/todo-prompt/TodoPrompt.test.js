import React from 'react'
import renderer from 'react-test-renderer'

import TodoPromp from './TodoPrompt'

describe('TodoPromp', () => {
  const mockClosePrompt = jest.fn()
  const mockCreateTodo = jest.fn()
  const mockSetTodoDescription = jest.fn()
  const testTodoDescription = 'Loren Ipsum'

  const todoPromptCmp = renderer.create(
    <TodoPromp
      closePrompt={mockClosePrompt}
      createTodo={mockCreateTodo}
      todoDescription={testTodoDescription}
      setTodoDescription={mockSetTodoDescription}
    />
  )

  it('renders correctly', () => {
    const snapShot = todoPromptCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('triggers create todo', () => {
    const e = { preventDefault: jest.fn() }
    const createBtn = todoPromptCmp.root.findByProps({
      'data-testid': 'create-todo',
    })
    createBtn.props.onSubmit(e)
    expect(mockCreateTodo).toHaveBeenCalled()
  })

  it('triggers setDescription when changing input', () => {
    const e = {
      target: {
        value: 'test',
      },
    }
    const input = todoPromptCmp.root.findByProps({
      'data-testid': 'todo-description-input',
    })
    input.props.onChange(e)
    expect(mockSetTodoDescription).toHaveBeenCalled()
  })

  it('triggers closePrompt when clicking overlay', () => {
    const overlay = todoPromptCmp.root.findByProps({
      'data-testid': 'close-prompt',
    })
    overlay.props.onClick()
    expect(mockClosePrompt).toHaveBeenCalled()
  })
})
