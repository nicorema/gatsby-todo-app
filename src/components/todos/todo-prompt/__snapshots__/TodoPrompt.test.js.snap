// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoPromp renders correctly 1`] = `
Array [
  <div
    className="sc-gtssRu iupHtr"
    data-testid="close-prompt"
    onClick={[MockFunction]}
  />,
  <form
    className="sc-dlnjPT jaFted"
    data-testid="create-todo"
    onSubmit={[Function]}
  >
    <label
      className="css-1nncrbt"
      htmlFor="description"
    >
      Información del Todo
    </label>
    <textarea
      className="css-16gg0af"
      data-testid="todo-description-input"
      id="description"
      onChange={[Function]}
      placeholder="Descripción del Todo"
      required={true}
      value="Loren Ipsum"
    />
    <div
      className="sc-bdnylx fXrGza"
    >
      <button>
        Confirmar
      </button>
    </div>
  </form>,
]
`;
